tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -762,11 +766,20 @@ def number_peaks(x, n):
     :return: the value of this feature
     :return type: float
     """
-    res = list()
+    x = np.asarray(x)
+    x_reduced = x[n:-n]
+
+    res = None
     for i in range(1, n + 1):
-        res += [(x > np.roll(x, i))[n:-n]]
-        res += [(x > np.roll(x, -i))[n:-n]]
-    return sum(reduce(lambda a, b: a & b, res))
+        result_first = (x_reduced > np.roll(x, i)[n:-n])
```

Comment:
The method stored each temporary result of the > operation in a list and reduced them at once in the end. This is not needed as the operation can be done immediately.
The [n:-n] and the > operation are actually quite costly (I have profiled it), so I tried to reduce the amount of times we have to do this.